### PR TITLE
Documented blocks storage limitation

### PR DESCRIPTION
**Pull request by pracucci**
**What this PR does**:
I've documented the known issue related to the issue #2366.

**Which issue(s) this PR fixes**:
N/A

**Checklist**
- [ ] Tests updated
- [x] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/operations/blocks-storage.md [hunk 1]
```diff
@@ -455,6 +455,20 @@ compactor:
 
 ## Known issues
 
+### Can't ingest samples older than 1h compared to the latest received sample of a tenant
+
+The blocks storage opens a TSDB for each tenant in each ingester receiving samples for that tenant. The received series are kept in the TSDB head and then a new block is cut from the head once the time range period covered by the series in the head reaches the block range period (defaults to 2h).
+
+When a new block is cut from head, the new block contains samples up until -50% of the configured block range period. This means that if you use the default 2h block range period, the head will contain up until 3h series (2h + 50%) and when a new block is cut from head it will contain series up until 1h ago (2h - 50%); technically it's 1h less than the max timestamp in the head.
+
+Given TSDB doesn't allow to append samples out of head bounds, the Cortex blocks storage can't ingest samples older than 1h compared to the latest received sample of a tenant.
```
Comment (pstibrany) [inline docs/operations/blocks-storage.md]: Limit is computed for each push as Max of (minT, maxT-0.5*block range), where minT and maxT are minimum/maximum sample times in the head.

When head covers less than 0.5 of block range (<1h for 2h block range), samples cannot be older than min time in the head.

When head covers more than 0.5 of block range (>1h for 2h block range), samples cannot be older than half block range since max time in the head.

In other words, it can be less than 1h :-(

File: docs/operations/blocks-storage.md [hunk 2]
```diff
@@ -455,6 +455,20 @@ compactor:
 
 ## Known issues
 
+### Can't ingest samples older than 1h compared to the latest received sample of a tenant
+
+The blocks storage opens a TSDB for each tenant in each ingester receiving samples for that tenant. The received series are kept in the TSDB head and then a new block is cut from the head once the time range period covered by the series in the head reaches the block range period (defaults to 2h).
+
+When a new block is cut from head, the new block contains samples up until -50% of the configured block range period. This means that if you use the default 2h block range period, the head will contain up until 3h series (2h + 50%) and when a new block is cut from head it will contain series up until 1h ago (2h - 50%); technically it's 1h less than the max timestamp in the head.
```
Comment (pstibrany) [inline docs/operations/blocks-storage.md]: Different description of same idea...

New block is cut from the head, when head (in-memory data) covers more than 1.5x of the block range. For 2 hours block range, it means that head needs to have 3h of data to cut a block. Block "start time" is always the minimum sample time in the head, while block "end time" is aligned on block range boundary. New block always covers single "block range" period. Data stored into a block is then removed from the Head. That means that after cutting the block, head will already have at least 1h of data, but possibly even more.

Comment (pstibrany) [reply docs/operations/blocks-storage.md]: > and when a new block is cut from head it will contain series up until 1h ago (2h - 50%)

When I read this few several times, I understood "it" as "block", but in fact it refers to the head.

File: docs/operations/blocks-storage.md (hunk 1 above)
Comment (pstibrany) [reply docs/operations/blocks-storage.md]: > In other words, it can be less than 1h :-(

This is true for completely new tenants. For normal operation, it is usually 1h.

Comment (pracucci) [reply docs/operations/blocks-storage.md]: > When head covers less than 0.5 of block range (<1h for 2h block range), samples cannot be older than min time in the head.

This is true, but looks more an edge case compared to the general case I'm trying to describe here.

Comment (pstibrany) [reply docs/operations/blocks-storage.md]: > This is true, but looks more an edge case compared to the general case I'm trying to describe here.

Right. I think I was mostly writing note to better understand this.

File: docs/operations/blocks-storage.md (hunk 2 above)
Comment (pracucci) [reply docs/operations/blocks-storage.md]: I picked your suggestion and I've done few changes. May you take a look now, please?
